### PR TITLE
Add `major-version-check` action

### LINKED ISSUE #8994: Prevent unintended major releases
## Problem

Major is currently the default version type when creating a changeset. This makes it too easy for folks to accidentally target a major version with changes that are only patch or minor. Most instances are caught before PR merge, but if they slip into Version Packages the noise in the description makes it easy for major versions of non-critical packages to ship as a major version release, usually `@shopify/polaris-icons` 🥲 . 

## Solution

- [ ] Reverse order of changeset version question from major, minor, patch to patch, minor, major
- [ ] Add a CI check that requires an extra step to move forward with targeting a major version (similar to change log check)

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHAT is this pull request doing?

This PR introduces a check to fail CI if new or modified changeset entries include a `major` version upgrade. These upgrades are easy to slip through the cracks, whereas, we expect them to be a very intentional decision. Authors can override this check by applying a `🤖Skip Major Check` label to their PR.

### CODE REVIEW

**Review by alex-page — APPROVED** [review]

File: .changeset/shaggy-sloths-knock.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'polaris-for-vscode': major
```
Comment (aaronccasanova) [inline .changeset/shaggy-sloths-knock.md]:
```suggestion
'polaris-for-vscode': minor
```

File: .changeset/strong-paws-camp.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-cli': major
```
Comment (aaronccasanova) [inline .changeset/strong-paws-camp.md]:
```suggestion
'@shopify/polaris-cli': minor
```

File: .github/workflows/major-version-check.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          if [[ -z "$changeset_files" ]]; then
+            echo "No changeset files detected"
+            exit 0
+          fi
+
+          if ! echo "$changeset_files" | xargs grep -qE "^\s*[\"'].*?[\"']\s*:\s*major\s*$"; then
+            echo "No major version upgrade detected"
+            exit 0
+          else
+            echo "This PR contains a major version upgrade."
```
Comment (alex-page) [inline .github/workflows/major-version-check.yml]:
```suggestion
            echo "This PR contains a major version upgrade which is only used for breaking changes."
```

**Review by chloerice — APPROVED** [review]
Thanks for adding this Aaron!! :shipit:

**Pull request merged**